extendr-api/src/error.rs

```diff
@@ -170,6 +172,8 @@ impl std::fmt::Display for Error {
                 write!(f, "It is only possible to return a reference to self.")
             }
             Error::NoGraphicsDevices(_robj) => write!(f, "No graphics devices active."),
+            // this is very unlikely to occur, and it would just say: Rust error: could not convert slice to array
+            Error::TryFromSliceError(std_error) => write!(f, "Rust error: {}", std_error),
```

Comment:
I think we could improve this error message. Particularly if we want to encourage users to use arrays in function arguments. 

```rs
#[extendr]
/// @export
fn round_trip_array_f64(x: [f64; 4]) -> [f64; 4] {
    x
}
```

Supplying this with a vector length < 4 returns the error 
```
Error in `round_trip_array_f64()`:
! could not convert slice to array
```
which isn't clear from an R user perspective.

Reply:
I'll fix that. I didn't think that error would ever appear.. I don't know why I thought that.. I am sorry for the obvious fail here.

Reply:
All good! Do you think we can do a length check or something similar?

Reply:
Yeah. Let me get back to you. I might have lost a commit because I could swear I did this... Or something similar... 😅

I'll ping you later on this.